core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java

```diff
@@ -166,14 +166,18 @@
     private final Session session;
     private final ConnectorSession connectorSession;
     private final Map<NodeRef<Expression>, Type> expressionTypes;
+    private final Map<NodeRef<Expression>, ResolvedFunction> resolvedFunctions;
     private final InterpretedFunctionInvoker functionInvoker;
     private final TypeCoercion typeCoercion;
 
     // identity-based cache for LIKE expressions with constant pattern and escape char
     private final IdentityHashMap<LikePredicate, JoniRegexp> likePatternCache = new IdentityHashMap<>();
     private final IdentityHashMap<InListExpression, Set<?>> inListCache = new IdentityHashMap<>();
 
-    public ExpressionInterpreter(Expression expression, PlannerContext plannerContext, Session session, Map<NodeRef<Expression>, Type> expressionTypes)
+    public ExpressionInterpreter(Expression expression,
+            PlannerContext plannerContext,
+            Session session, Map<NodeRef<Expression>, Type> expressionTypes,
+            Map<NodeRef<Expression>, ResolvedFunction> resolvedFunctions)
```

Comment:
@dain  please review

Reply:
@dain need your help to review when you have free time.

Reply:
+ This commit have not affect the TestDeltaTaskFailureRecoveryTest
+ I have rerun the test case (io.trino.plugin.deltalake.TestDeltaTaskFailureRecoveryTest) in my dev environment, these test cases can be successful as following.
```
[INFO] Results:
[INFO]
[INFO] Tests run: 19, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  28:01 min
[INFO] Finished at: 2022-05-09T01:00:03+08:00
[INFO] ------------------------------------------------------------------------
```